contracts/src/SequencerInbox.sol

```diff
@@ -74,12 +77,17 @@ contract SequencerInbox is ISequencerInbox, Initializable, UUPSUpgradeable, Owna
         uint256[] calldata contexts,
         uint256[] calldata txLengths,
         uint256 firstL2BlockNumber,
+        uint256 txBatchVersion,
         bytes calldata txBatch
     ) external override whenNotPaused {
         if (msg.sender != sequencerAddress) {
             revert NotSequencer(msg.sender, sequencerAddress);
         }
 
+        if (txBatchVersion != currentTxBatchVersion) {
```

Comment:
Not the way I understand, because it's not necessarily true that the sequencer client code is submitting the right serialization version. Practically, it would be pretty hard, but if we did update the contracts but not sequencer we want to make sure we're not getting an unexpected batch format being sent

Reply:
We don't verify the batch format here, so a sequencer could still submit an invalid format as long as the version is correct - we validate the data later with the challenge game. But a sanity check here to detect old sequencers might still make sense.